### PR TITLE
[INLONG-5210][Sort] Add reporting metrics for Elasticsearch 7 connector and audit SDK

### LINKED ISSUE #5210: [Feature][Sort] Add reporting metrics for Elasticsearch 7 connector and audit SDK
### Description

Add reporting metrics for Elasticsearch 7 connector

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Oneal65**
- Fixes #5210 

### Motivation

Add reporting metrics for Elasticsearch 7 connector and audit SDK

### Modifications

Elasticsearch 7 connector

### CODE REVIEW

File: inlong-sort/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/table/RowElasticsearchSinkFunction.java [hunk 1]
```diff
@@ -119,6 +148,7 @@ private void processUpsert(RowData row, RequestIndexer indexer) {
         if (sinkMetricData.getNumBytesOut() != null) {
             sinkMetricData.getNumBytesOut().inc(document.length);
         }
+        outputMetricForAudit(document.length);
```
Comment (thesumery) [inline inlong-sort/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/table/RowElasticsearchSinkFunction.java:151]: Here only output +I/+U data？

File: inlong-sort/sort-connectors/elasticsearch-base/pom.xml [hunk 2]
```diff
@@ -152,6 +152,11 @@
             <artifactId>log4j-core</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.apache.inlong</groupId>
+            <artifactId>audit-sdk</artifactId>
+            <version>1.3.0-SNAPSHOT</version>
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/elasticsearch-base/pom.xml:158]: It is better to use ${xxx.version} to replace it.

File: inlong-sort/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/table/RowElasticsearchSinkFunction.java (hunk 1 above)
Comment (Oneal65) [reply inlong-sort/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/table/RowElasticsearchSinkFunction.java:151]: @thesumery should  the rowData of DELETE mode be counted?

File: inlong-sort/sort-connectors/elasticsearch-base/pom.xml (hunk 2 above)
Comment (Oneal65) [reply inlong-sort/sort-connectors/elasticsearch-base/pom.xml:158]: DONE

**Review by gong — APPROVED** [review]

**Review by EMsnap — APPROVED** [review]

**Pull request merged**